File: c2/c2.py
def createFrame(names):
    maxLength = len(max(names, key=len))
    width = maxLength + 4
    frame = ["*" * width]
    for name in names:
        frame.append(f"* {name.ljust(maxLength)} *")
    frame.append("*" * width)
    return "\n".join(frame)

File: c2/c2.ts
function createFrame(names: string[]): string {
  const maxLength: number = Math.max(...names.map(name => name.length));
  const width: number = maxLength + 4;
  const frame: string[] = ['*'.repeat(width)];
  for (const n of names) {
    frame.push(`* ${n.padEnd(maxLength, ' ')} *`);
  }
  frame.push('*'.repeat(width));
  return frame.join('\n');
}
